packages/frontend-plugin-api/src/components/ExtensionBoundary.tsx

```diff
@@ -15,15 +15,35 @@
  */
 
 import React, { ReactNode } from 'react';
+import { AnalyticsContext } from '@backstage/core-plugin-api';
 import { BackstagePlugin } from '../wiring';
+import { RouteRef } from '../routing';
+import { ErrorBoundary } from './ErrorBoundary';
+import { toInternalRouteRef } from '../routing/RouteRef';
 
 /** @public */
 export interface ExtensionBoundaryProps {
-  children: ReactNode;
+  id: string;
   source?: BackstagePlugin;
+  routeRef?: RouteRef;
+  children: ReactNode;
 }
 
 /** @public */
 export function ExtensionBoundary(props: ExtensionBoundaryProps) {
-  return <>{props.children}</>;
+  const { id, source, routeRef, children } = props;
+
+  const attributes = {
+    extension: id,
+    pluginId: source?.id,
+    routeRef: routeRef
+      ? toInternalRouteRef(routeRef).getDescription()
```

Comment:
@Rugvip is this the right way of getting the id?

Reply:
Nah it might also be a random stack trace line. Tbh I think we should skip this attribute in the new system. The extension ID should be plenty information anyway and provide much more insight compared to the existing system.